### PR TITLE
libevmcore, libevm: turn enum Tier into an enum class

### PR DESCRIPTION
**Pull request by pirapira**
Just turning `enum Tier` into `enum class Tier`.

I made a similar change in Solidity, and learned that `cpp-ethereum` has the same code: https://github.com/ethereum/solidity/pull/1562

### CODE REVIEW

File: libevmcore/Instruction.h [hunk 1]
```diff
@@ -228,7 +228,7 @@ inline Instruction logInstruction(unsigned _number)
 	return Instruction(unsigned(Instruction::LOG0) + _number);
 }
 
-enum Tier
+enum class Tier : unsigned
```
Comment (chfast) [inline libevmcore/Instruction.h]: Can you also rename the elements by removing the `Tier` suffix?

Comment (pirapira) [inline libevmcore/Instruction.h]: That's a good idea.

**Comment by gcolvin** [comment]
#3479 is merged, so you are good to go @pirapira, except that there are conflicts now, as I feared.

**Comment by pirapira** [comment]
I solved conflicts and took in @chfast 's suggestion.

**Comment by chfast** [comment]
LGTM

**Pull request merged**